updated plot function with log transformation ~/R/plot.R

alfredojavier55 opened the pull request:
Hi Stefan, I'm happy to write you. I have some time today and had some time with Thomas at the conference. I update 12 lines on the plot function adding an argument called log_transform. This allows to change the y axis to log. This is handy in our phageEX when dealing with large populations as phages, maybe could be used if vectors are also modelled. I haven't update the others ## ' comments and files to roxygenise. Maybe you can evaluate this first try and then work a little more on the other files. 
I used this simple example to see, how it looks like. Maybe we could write a simple phage example, to be more aplicable to real world examples. 

# Example usage of the modified plot function
model <- SIR(u0 = data.frame(S = rep(10000, 10),
                             I = rep(10, 10),
                             R = rep(10, 10)),
             tspan = 1:24,
             beta = 0.8,
             gamma = 0.077)
result <- run(model)

# Test the updated plot function with log transformation
par(mfrow = c(2, 1))
plot(result, log_transform = FALSE) 
plot(result, log_transform = TRUE)
par(mfrow = c(1, 1))

stewid commented:
Thanks Alfredo. I like your suggestion to make it possible to log-transform the data. I'll take a closer look and review the code.

stewid commented on R/plot.R:
```diff
@@ -494,7 +494,7 @@ setMethod(
     signature(x = "SimInf_model", y = "ANY"),
     function(x, y, level = 1, index = NULL, range = 0.5, type = "s",
-             lwd = 2, frame.plot = FALSE, legend = TRUE, ...) {
+             lwd = 2, frame.plot = FALSE, legend = TRUE, log_transform = FALSE, ...) { #ajab
```
Could you please remove the trailing `#ajab` from the lines.

stewid replied:
I suggest that we change the parameter name from `log_transform=FALSE` to `log=""` for consistency with the R plot function https://search.r-project.org/R/refmans/graphics/html/plot.default.html

This would also enable log-values of the x-axis.

stewid reviewed (commented):
I've made some suggestions for changes.

alfredojavier55 commented:
Hi Stefan, I removed the trailing lines, then used the log from base R. There were some problems in the arguments, the plot was behaving correctly but a warning appared: "plot(result)
> Error in plot.window(...) : "log=" specification must be character. . "

I solved them making modifications in two places, plot_method and plot_data:

- plot() Method for SimInf_model:
Added a default value for the log argument (log = "") to ensure it behaves like the base R plot() function.
Ensured the log argument is passed explicitly to init_plot_argv() and plot_data().

- plot_data() Function:
Extracted the log argument from argv and passed it explicitly to the plot() function.
Removed the log argument from argv to prevent it from being passed to lower-level functions like lines() or polygon().

I did not update the vignettes:
"Argument names in code not in docs:
    log
  Mismatches in argument names:
    Position: 10 Code: log Docs: ... "

Hope this helps!

- Testing: 
# Example usage of the modified plot function
model <- SIR(u0 = data.frame(S = rep(10000, 10),
                             I = rep(10, 10),
                             R = rep(10, 10)),
             tspan = 1:100,
             beta = 0.8,
             gamma = 0.077)

result <- run(model)
trajectory(result)
# Test the updated plot function with log transformation

plot(result)

par(mfrow = c(4, 2))
plot(result, , main = "Base plot")
plot(result, log = "y", main = "Log Y-Axis") 
plot(result, log = "x", main = "Log X-Axis") 
plot(result, log = "y", ylim = c(1, 10000), main = "Log Y-Axis and limits")
plot(result, log = "xy", ylim = c(1, 10000), main = "Logarithmic X&Y-Axis")
plot(result, log = "x", ylim = c(1, 20), main = "Log X-Axis and limits")
plot(result, log = "y", ylim = c(1, 500), main = "Log Y-Axis and limits")
plot(result, ~ S, log = "y", ylim = c(1, 10000), main = "S")
par(mfrow = c(1, 1))

![Screenshot 2025-05-13 123958](https://github.com/user-attachments/assets/96da21e0-7806-451c-b936-189b0ded2c3b)

stewid commented:
Hi, thanks for the update. The `SimInf.Rcheck` folder has accidentally been included in the commit. Could you please have a look at that, and then I'll take a look on your update.

alfredojavier55 commented:
Dear Stefan, thanks for checking. The folder is deleted. Hope you can see the changes more clearly now. Thanks.

stewid commented:
Thanks @alfredojavier55 for the update, I will check later this week.

stewid commented on R/plot.R:
```diff
@@ -309,6 +309,11 @@ init_plot_argv <- function(model, compartments, pd, type, lwd, ...) {
+    # Remove log from argv to avoid passing it to lower-level functions
+    argv$log <- NULL  
+
     ## Plot lines
```
I suggest we simplify this. If you add `log` as parameter to the function (`plot_data <- function(pd, argv, lty, col, frame.plot, legend, log) {`), we can remove the lines 312-316.

alfredojavier55 replied:
I agree, we don't need extra lines.

stewid commented on R/plot.R:
```diff
@@ -317,7 +322,7 @@ plot_data <- function(pd, argv, lty, col, frame.plot, legend) {
         if (i == 1) {
             argv$frame.plot <- frame.plot
-            do.call(plot, argv)
+            do.call(plot, c(argv, list(log = log_arg)))
```
After adding `log` as a parameter, you can instead write `c(argv, log = log)`.

alfredojavier55 replied:
I updated as suggested

stewid commented on R/plot.R:
```diff
@@ -510,7 +515,7 @@ setMethod(
         }
 
-        argv <- init_plot_argv(x, y, pd, type, lwd, ...)
+        argv <- init_plot_argv(x, y, pd, type, lwd, log = log, ...)  
```
The log argument is not used in this function, so you can remove `log = log,` here and instead pass the log argument to the `plot_data` function below instead.

alfredojavier55 replied:
Dear Stefan, thanks for the suggestions, much more simple and straigh now. I did check the examples and works well.

alfredojavier55 replied:
Dear @stewid,  could you please review or merge this PR when you have a chance? Thanks! I also resolved the conversation.

stewid replied:
@alfredojavier5 I apologize for the delay. I'll try to find time next week.

alfredojavier55 replied:
@stewid no worries, greetings from Greifswald!

alfredojavier55 commented:
Good morning @stewid, yesterday's changes were updated.

stewid commented on ult)  # Should work without specifying log:
```diff
@@ -0,0 +1,1202 @@
+Vignettes with name or keyword or title matching ‘plot’ using regular
```
Hi, thanks for the update. I'm not sure what this file is but it has accidentally been included in the commit. Could you please have a look at that, and then I'll take a look on your update.

alfredojavier55 replied:
The file was deleted. Thank you.

stewid reviewed (commented):
Hi, thanks for the update. Could you please remove the file that has accidentally been included in the commit.

stewid commented:
@alfredojavier55 Thanks
